zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/DocumentenApiPlugin.kt

```diff
@@ -222,40 +223,48 @@ class DocumentenApiPlugin(
 
     @PluginEvent(invokedOn = [EventType.CREATE, EventType.UPDATE])
     fun onSave() {
-        if (apiVersion != null && !API_VERSIONS.contains(apiVersion)) {
+        if (apiVersion != null && !documentenApiVersionService.isValidVersion(apiVersion!!)) {
             throw ValidationException("Unknown API version '$apiVersion'.")
         }
     }
 
     private fun storeDocument(
         execution: DelegateExecution,
-        metadata: Map<String, Any>,
-        title: String,
-        confidentialityLevel: String?,
-        status: DocumentStatusType,
-        language: String?,
-        filename: String?,
-        contentAsInputStream: InputStream,
-        description: String?,
-        informationObjectType: String,
+        metadata: Map<String, Any?>,
+        titel: String?,
+        vertrouwelijkheidaanduiding: String?,
+        status: DocumentStatusType?,
+        taal: String?,
+        bestandsnaam: String?,
+        inhoudAsInputStream: InputStream,
+        beschrijving: String?,
+        informatieobjecttype: String?,
         storedDocumentUrl: String,
     ) {
+        val vertrouwelijkheidaanduidingEnum = Vertrouwelijkheid.fromKey(
+            vertrouwelijkheidaanduiding ?: getMetadataField(
+                metadata,
+                VERTROUWELIJKHEIDAANDUIDING_FIELD
+            )
+        )
+
         val request = CreateDocumentRequest(
             bronorganisatie = bronorganisatie,
-            creatiedatum = getLocalDateFromMetaData(metadata, "creationDate", LocalDate.now())!!,
-            titel = title,
-            vertrouwelijkheidaanduiding = Vertrouwelijkheid.fromKey(confidentialityLevel),
-            auteur = metadata["author"] as String? ?: DEFAULT_AUTHOR,
-            status = status,
-            taal = language ?: DEFAULT_LANGUAGE,
-            bestandsnaam = filename,
+            creatiedatum = getLocalDateFromMetaData(metadata, CREATIEDATUM_FIELD) ?: LocalDate.now(),
+            titel = titel ?: getMetadataField(metadata, TITEL_FIELD)!!,
```

Comment:
This used to fall back onto `fileNameNotNull`, and now it doesn't anymore. I guess it should now fall back to `bestandsnaam ?: getMetadataField(metadata, BESTANDSNAAM_FIELD)`

Reply:
I've added all variations to the `TITEL_FIELD`